### PR TITLE
Dependency and pom updates

### PR DESCRIPTION
**Pull request by adgonz90**
Updated Dependency versions to resolve vulnerabilities.
Updated plugin configurations and removed old plugins.
Added Maven Wrapper for consistent builds and updated CircleCI to use it.
Updated Archetypes to have correct versions on release.

### CODE REVIEW

**Review by Linesmerrill — COMMENTED** [review]
Inline comments below. CI is green and the overall direction (archetype tokenization, Maven wrapper, dep hygiene, modern native/spotbugs plugins) looks good — leaving these as questions/suggestions before approval.

File: pom.xml [hunk 1]
```diff
@@ -612,29 +630,21 @@
...
-                <version>2.10.1</version>
-            </dependency>
-            <!-- CVE-2025-48976 (High/7.5) - Fixed in 1.6.0, added 2026-02-11
-                 TODO: Remove this override when spring-boot-dependencies includes commons-fileupload >= 1.6.0
-                 Check: mvn dependency:tree | grep commons-fileupload -->
-            <dependency>
-                <groupId>commons-fileupload</groupId>
-                <artifactId>commons-fileupload</artifactId>
-                <version>1.6.0</version>
+                <version>2.14</version>
```
Comment (Linesmerrill) [inline pom.xml]: issue: `2.14` is not a published version of `commons-configuration2` on Maven Central — only `2.14.0` exists (verified against `repo.maven.apache.org/maven2/org/apache/commons/commons-configuration2/maven-metadata.xml`). Maven's version range parsing may coerce `2.14` to something unexpected, or modules without a local cache could fail to resolve.

```suggestion
                <version>2.14.0</version>
```

File: pom.xml [hunk 2]
```diff
@@ -612,29 +630,21 @@
                 <artifactId>testcontainers</artifactId>
                 <version>1.21.2</version>
             </dependency>
-            <!-- CVE-2026-24400 (High/8.2) - Fixed in 3.27.7, added 2026-02-10
-                 TODO: Remove this override when spring-boot-dependencies includes assertj-core >= 3.27.7
-                 Check: mvn dependency:tree | grep assertj-core -->
+            <!-- CVE-2025-48924 (High/7.5) - Fixed in 3.18.0, added 2025-07-11
```
Comment (Linesmerrill) [inline pom.xml]: nit: comment says "Fixed in 3.18.0" but the override pins `3.20.0`. Easy to misread as a typo when scanning the override list later.

```suggestion
            <!-- CVE-2025-48924 (High/7.5) - Fixed in 3.18.0; pinned to 3.20.0, added 2025-07-11
```

File: pom.xml [hunk 3]
```diff
@@ -612,29 +630,21 @@
...
-            <!-- CVE-2025-48976 (High/7.5) - Fixed in 1.6.0, added 2026-02-11
-                 TODO: Remove this override when spring-boot-dependencies includes commons-fileupload >= 1.6.0
-                 Check: mvn dependency:tree | grep commons-fileupload -->
-            <dependency>
-                <groupId>commons-fileupload</groupId>
-                <artifactId>commons-fileupload</artifactId>
-                <version>1.6.0</version>
+                <version>2.14</version>
             </dependency>
             <!--./END Temporary For Vulnerability Fix-->
```
Comment (Linesmerrill) [inline pom.xml:657]: question: the previous `commons-fileupload` override (CVE-2025-48976, pinned to 1.6.0) was dropped in this hunk. Has `./mvnw dependency:tree | grep commons-fileupload` been run on a service module to confirm Spring Boot 3.5.14's BOM brings ≥1.6.0 transitively? If not, the CVE override silently regresses.

File: pom.xml [hunk 4]
```diff
@@ -923,19 +930,18 @@
                 <plugin>
                     <groupId>org.sonatype.central</groupId>
                     <artifactId>central-publishing-maven-plugin</artifactId>
-                    <version>0.9.0</version>
+                    <version>0.10.0</version>
                     <extensions>true</extensions>
                     <configuration>
                         <publishingServerId>central</publishingServerId>
                         <autoPublish>true</autoPublish>
```
Comment (Linesmerrill) [inline pom.xml:945]: question: `<waitUntil>published</waitUntil>` was removed from the central-publishing config. Intentional? Without it, the release job finishes before the artifact is actually live on Maven Central — fine if deliberate (e.g. CI was timing out on publish), just want to confirm it wasn't an accidental drop while bumping the plugin to 0.10.0.

File: function/function-samples/sample-function-rest-book/pom.xml [hunk 5]
```diff
@@ -82,46 +89,6 @@
             <plugin>
                 <groupId>org.jetbrains.kotlin</groupId>
                 <artifactId>kotlin-maven-plugin</artifactId>
```
Comment (Linesmerrill) [inline function/function-samples/sample-function-rest-book/pom.xml]: question: this module has no `.kt` source files — only Java under `src/main/java`. With the previous `<executions>` removed and `kotlin-stdlib` newly added as a runtime dep, is either piece actually needed here?

If `spring-cloud-function-adapter-gcp` (or another transitive) requires `kotlin-stdlib` at runtime, the dep is fine — but the plugin block can probably be dropped entirely:

```suggestion
```

(empty suggestion = remove the `<plugin>` block)

File: mvnw [hunk 6]
```diff
@@ -0,0 +1,39 @@
+#!/bin/sh
```
Comment (Linesmerrill) [inline mvnw:1]: suggestion: this is a hand-rolled wrapper (39 lines). The canonical script produced by `mvn wrapper:wrapper` is significantly more battle-tested across edge cases (proxies, alternate `JAVA_HOME`, Windows quoting via `mvnw.cmd`, etc.). If there's no specific reason to roll your own, consider regenerating with the official plugin so future Maven wrapper bumps are a one-liner.

File: pom.xml (hunk 4 above)
Comment (adgonz90) [reply pom.xml:945]: yes, CircleCI was timing out on publish.  Sonatype can take more than an hour to complete the bundle release.

File: pom.xml (hunk 2 above)
Comment (adgonz90) [reply pom.xml]: updated

File: pom.xml (hunk 3 above)
Comment (adgonz90) [reply pom.xml:657]: i dont think its being used in Synapse, but added back for safety.

File: function/function-samples/sample-function-rest-book/pom.xml (hunk 5 above)
Comment (adgonz90) [reply function/function-samples/sample-function-rest-book/pom.xml]: removed.

File: mvnw (hunk 6 above)
Comment (adgonz90) [reply mvnw:1]: updated!

File: pom.xml (hunk 4 above)
Comment (Linesmerrill) [reply pom.xml:945]: perfect! thanks!

**Review by Linesmerrill — APPROVED** [review]
